### ISSUE TITLE
[gen-l10n] wrong generation when the placeholder is missing in other languages

### ISSUE THREAD

**Issue by GoShikhar** [flutter/flutter#119557]
Post 3.7 upgrade, getting the following error for all the i10n strings.

```
Error: Can't declare a member that conflicts with an inherited one.
.dart_tool/…/gen_l10n/app_localizations_ur.dart:17
  String get devOptionsEnabled => '';
             ^^^^^^^^^^^^^^^^^
: Context: This is the inherited member.
.dart_tool/…/gen_l10n/app_localizations.dart:140
  String devOptionsEnabled(String appName);
         ^^^^^^^^^^^^^^^^^
: Error: Can't declare a member that conflicts with an inherited one.
.dart_tool/…/gen_l10n/app_localizations_bn.dart:449
  String get pulseChlngAcc => '';

             ^^^^^^^^^^^^^
: Context: This is the inherited member.
.dart_tool/…/gen_l10n/app_localizations.dart:1004
  String pulseChlngAcc(String player);

         ^^^^^^^^^^^^^
Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.

* Where:
Script 'C:\Users\Shikhar\Documents\Github\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1151
```

The code gen takes place after running the command _flutter gen-l10n_ as [per this guide](https://docs.flutter.dev/development/accessibility-and-localization/internationalization).

Things I've tried:
1. Uninstall flutter, reinstall 3.7 fresh
2. Flutter clean

No code changes in project since the upgrade.

**Comment by huycozy** [comment]
@GoShikhar I can not reproduce this issue from my end. Can you please share your `arb` file content (just a snippet to reproduce) along with the output of `flutter doctor -v`?

**Comment by M123-dev** [comment]
While migrating, I experienced the same and I think I figured out what the problem is:

The String in my app_en.arb:

```
    "confirm_clear_user_list": "You're about to clear this list ({name}): are you sure you want to continue?",
    "@confirm_clear_user_list": {
        "description": "Asking about whether to clear the list or not",
        "placeholders": {
            "name": {
                "type": "String"
            }
        }
    },
```

The same in app_ja.arb

```
    "confirm_clear_user_list": "すべての履歴を削除しようとしています。続けていいですか。",
    "@confirm_clear_user_list": {
        "description": "Asking about whether to clear the list or not",
        "placeholders": {
            "name": {
                "type": "String"
            }
        }
    },
```


app_localizations_en.dart

``` dart
  @override
  String confirm_clear_user_list(String name) {
    return 'You\'re about to clear this list ($name): are you sure you want to continue?';
  }
```


app_localizations_ja.dart

``` dart
  @override
  String get confirm_clear_user_list => 'すべての履歴を削除しようとしています。続けていいですか。 clear list';
```


Since the translation isn't using the placeholder, it's for some reason generated as a normal getter without parameters even though a `String confirm_clear_user_list(String name)` is required in the abstract AppLocalizations

**Comment by dkbast** [comment]
Can reproduce it - the problem happens if we have a placeholder which is not part of the message - This should be allowed, as we sometimes want to give the translation team the option to use additional information, but can't know for sure if this will happen

**Comment by huycozy** [comment]
@M123-dev @dkbast Is the build also failed from your end? 

I tested with string which @M123-dev provided above (`confirm_clear_user_list`). There is no error from the console but there is an error from generated `app_localization_ja.dart` file:

```
Class 'AppLocalizationsJa' can't define field 'confirm_clear_user_list' and have method 'AppLocalizations.confirm_clear_user_list' with the same name.
```

So, I think you need to update your `ja` string to be the same as `en` string (all have `name` placeholders inside string):

```
{
    "confirm_clear_user_list": "すべての履歴を削除しようとしています({name})。続けていいですか。",
    "@confirm_clear_user_list": {
        "description": "Asking about whether to clear the list or not",
        "placeholders": {
            "name": {
                "type": "String"
            }
        }
    }
}
```

Then the tool can generate fine:

```
  @override
  String confirm_clear_user_list(String name) {
    return 'すべての履歴を削除しようとしています($name)。続けていいですか。';
  }
```

**Comment by huycozy** [comment]
@dkbast I see the issue. Thanks for clarifying that. 

### Steps to reproduce
1. Setup a project as below sample code with dart and arb files. There is one language that contains a placeholder while another does not.
2. Generate localization file with command: `flutter gen-l10n`
3. Run the project
4. Check error from the console log

```
lib/l10n/app_localizations_ja.dart:8:14: Error: Can't declare a member that conflicts with an inherited one.
  String get confirm_clear_user_list => 'すべての履歴を削除しようとしています。続けていいですか。';
             ^^^^^^^^^^^^^^^^^^^^^^^
lib/l10n/app_localizations.dart:102:10: Context: This is the inherited member.
  String confirm_clear_user_list(String name);
         ^^^^^^^^^^^^^^^^^^^^^^^
Failed to compile application.
```

### Analysis

This issue is reproducible on the latest `stable` and `master` channels whilst it works as expected on Flutter `3.3.10`. 

| 3.7.1/3.8.0 | 3.3.10 |
| --------------- | --------------- | 
✅ | ❌ | 

✅: No Issue ❌: Issue reproduced


Comparing the generated localization `ja` files:
- `Stable/Master (3.7.1/3.8.0-2.0.pre.43)`:

```dart
  @override
  String get confirm_clear_user_list => 'すべての履歴を削除しようとしています。続けていいですか。';
```

- `3.3.10`:

```dart
  @override
  String confirm_clear_user_list(String name) {
    return 'すべての履歴を削除しようとしています。続けていいですか。';
  }
```

<details>
<summary>Sample code (dart+arb)</summary>

`app_en.arb`:

```dart
{
  "confirm_clear_user_list": "You're about to clear this list ({name}): are you sure you want to continue?",
  "@confirm_clear_user_list": {
    "description": "Asking about whether to clear the list or not",
    "placeholders": {
      "name": {
        "type": "String"
      }
    }
  }
}
```

`app_ja.arb`:

```dart
{
    "confirm_clear_user_list": "すべての履歴を削除しようとしています。続けていいですか。",
    "@confirm_clear_user_list": {
        "description": "Asking about whether to clear the list or not"
    }
}
```

`main.dart`:

```dart
import 'package:flutter/material.dart';
import 'package:flutter_localizations/flutter_localizations.dart';
import 'package:test_gen_l10n/l10n/app_localizations.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      localizationsDelegates: const [
        AppLocalizations.delegate,
        GlobalMaterialLocalizations.delegate,
        GlobalWidgetsLocalizations.delegate,
        GlobalCupertinoLocalizations.delegate,
      ],
      supportedLocales: const [
        Locale('ja', ''), // Japanese, no country code
      ],
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Test'),
        ),
        body: Column(
          children: [
            Text(AppLocalizations.of(context).confirm_clear_user_list('Mar')),
          ],
        ),
      ),
    );
  }
}
```

</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.7.1, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.7.1 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7048ed95a5 (11 hours ago), 2023-02-01 09:07:31 -0800
    • Engine revision 800594f1f4
    • Dart version 2.19.1
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Dolphin 2021.3.1.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio Dolphin 2021.3.1.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[!] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D                         • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.7.2 19H218
    • macOS (desktop)  • macos                                    • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)     • chrome                                   • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```

```bash
[!] Flutter (Channel master, 3.8.0-2.0.pre.43, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.8.0-2.0.pre.43 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3570cce58e (64 minutes ago), 2023-02-03 04:34:17 +0300
    • Engine revision 2871970337
    • Dart version 3.0.0 (build 3.0.0-196.0.dev)
    • DevTools version 2.21.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Users/huynq/Library/Java/JavaVirtualMachines/corretto-1.8.0_302/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment Corretto-8.302.08.1 (build 1.8.0_302-b08)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.

```
</details>

**Comment by pedromassango** [comment]
@AlexV525 @thkim1011 I see you guys made some changes on i18n recently, could you have a look at this and help with any tips to workaround or fix the issue?

**Comment by christopherfujino** [comment]
@thkim1011 assigning a p3 as it seems you're working on it, feel free to reprioritize.

**Comment by MadaFrog** [comment]
When I first Googled the problem with my error, literally nothing showed up. So, in case there aresome otherpoor guys. kike me, here is the error I got

```
Error (Xcode): .dart_tool/flutter_gen/gen_l10n/app_localizations_de.dart:197:14: Error: Can't declare a member that conflicts with an inherited one.
```

And by looking in this file at line `197` I found the problematic String and looked for it in all the .arb files. 

Apparently "String with some {param}" works and "String with some ($param)" also works. So when you run the `flutter gen-l10n` command, it doesn't give any error and it seems like all worked just fine, but then the build fails and gives that weird error. This happens because some files had the {param} version and others had the ($param) version.

Hope this helped, ad thanks @M123-dev for the fix! 😁
